elks/arch/i86/kernel/irqtab.S

```diff
@@ -378,10 +139,12 @@ _irqit:
 	push	%si
 	push	%di
 //
-//	Recover data segment
+//	Recover kernel data segment
+//	Was pushed by the CALLF of the dynamic handler
+//	TODO: BP is better for stack work
 //
-//	seg	cs
-	mov	%cs:ds_kernel,%ds
+	mov	%sp,%si
+	mov	%ss:8(%si),%ds
```

Comment:
I understand this change, makes routine more dependent on DS trampoline, just as pop %ds does below.
Change is elegant fix, but I hope you understand I have to maintain this code, not you. I really would prefer ds_kernel change to be in seperate PR. Has this been tested on ROM?

Reply:
Yes, it works well in ROM, and EMU86 helps a lot to check the new code.

Reply:
On a side note, sometimes I think it might be good idea to have mechanism in EMU86 where register dump is displayed on same lines (or directly above or below) for "deep inspection" of processor registers when running in 'c' continuous trace mode. This would help greatly instead of having to single-step through possible offending code.